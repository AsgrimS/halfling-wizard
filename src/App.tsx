import React from "react";

import GlobalStyle from "./globalStyles";
import GlobalFonts from "./fonts/globalFonts";
import Explorer from "./pages/explorer/Explorer";

const App: React.FC = () => {
  return (
    <div data-testid="app">
      <GlobalStyle />
      <GlobalFonts />
      <Explorer />
    </div>
  );
};

export default App;
