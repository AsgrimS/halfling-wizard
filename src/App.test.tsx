import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";

import App from "./App";

jest.mock("./pages/explorer/Explorer", () => () => (
  <div data-testid="mock-explorer" />
));

describe("App Core", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render an App core component", () => {
    render(
      <MockedProvider>
        <App />
      </MockedProvider>
    );
    const appElement = screen.getByTestId("app");
    expect(appElement).toBeInTheDocument();

    const explorerElement = screen.getByTestId("mock-explorer");
    expect(explorerElement).toBeInTheDocument();

    expect(appElement).toMatchSnapshot();
  });
});
