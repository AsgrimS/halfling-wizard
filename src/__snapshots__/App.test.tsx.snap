// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Core should render an App core component 1`] = `
<div
  data-testid="app"
>
  <div
    data-testid="mock-explorer"
  />
</div>
`;
